### ISSUE TITLE
Wrong construction of redirect uri when using url parameters

### ISSUE THREAD

**Issue by MichaelLorenzen** [jgraph/drawio#747]
### Preflight Checklist

* [x] I agree to follow the [Code of Conduct](https://github.com/jgraph/drawio/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

**Describe the bug**
If I use the url parameters gitlab=, gitlab-id= and mode=gitlab the redirect-uri is build the wrong way. I discovered this whith my personal gitlab installation, but the bug can be reproduced with gitlab.com

**To Reproduce**
1. Add an personal drawIO application on gitlab.com
2. Open draw.io with url parameter, for example: 
   https://draw.io?gitlab=https://gitlab.com&gitlab-id=[...]&mode=gitlab
3. Hit "Open an existing diagramm" and "Authorize"
4. The gitlab-page opens with "An error has occured" and "The The redirect URI included is not valid."

**Expected behavior**
On gitlab.com the authorization dialog should appear, so that draw.io gets the token.

**draw.io version (In the Help->About menu of the draw.io editor):**

- draw.io version 12.6.5

**Additional context**
- The url used to authorize in the steps above is:
https://gitlab.com/oauth/authorize?client_id=[...]&=api read_repository write_repository&redirect_uri=https://www.draw.io/?gitlab=https://gitlab.html&response_type=token&state=123, with 
-- **&redirect_uri=https://www.draw.io/?gitlab=https://gitlab.html**
- But it should be:
https://gitlab.com/oauth/authorize?client_id=5cdc018a32acddf6eba37592d9374945241e644b8368af847422d74c8709bc44&scope=api read_repository write_repository&redirect_uri=https://www.draw.io/gitlab.html&response_type=token&state=123, with
-- **redirect_uri=https://www.draw.io/gitlab.html**

**Comment by MichaelLorenzen** [comment]
Is there an workaround available to allow authentification with personal gitlabs in the current implemantation?

**Comment by thinkJD** [comment]
I run in the same problem. Is there a version with this feature working?

**Comment by vedmak3** [comment]
you can see  \draw\js\app.min.js line 10828
 `GitLabClient.prototype.authenticate=function(a,d){if(null==window.onGitLabCallback){var b=mxUtils.bind(this,function(){var c=!0;this.ui.showAuthDialog(this,!0,mxUtils.bind(this,function(g,f){var l=window.location.href,l=l.substring(0,l.lastIndexOf("/")),l=encodeURIComponent(l+"/gitlab.html");null!=window.open(DRAWIO_GITLAB_URL+"/oauth/authorize?client_id="+this.clientId+"&scope="+this.scope+"&redirect_uri="+l+"&response_type=token&state=123","gitlabauth")?window.onGitLabCallback=mxUtils.bind(this,`

**Comment by vincib** [comment]
Hi. I have the same issue on 14.6.0, and it looks like it comes from the **GitLabClient.js** file, in the function **GitLabClient.prototype.authenticate**, where the following JS is used to get the url root : 

`var dir = href.substring(0, href.lastIndexOf('/'));`

Which brings the right URL when using gitlab.com, but not when using your own. 
eg. from here : https://draw.octopuce.fr/index.html?gitlab=https://gitlab.octopuce.fr&gitlab-id=16cfb06db2a2da2b4a86abdeeda6c2d03fb7365792cdf591861d37a6435bbd38 
If I try to authorize myself on my gitlab, I have this "dir" variable value : 

`https://draw.octopuce.fr/index.html?gitlab=https:/`

which is obviously wrong. 

I replaced this dir line by mine here (feel free to use it, I will never claim any ownership on this crude lines :)  )
```
if (href.includes('?')) {
  var dir	= href.substring(0, href.indexOf('?'));
   dir = dir.substring(0, dir.lastIndexOf('/'));
} else {
  var dir = href.substring(0, href.lastIndexOf('/'));
}
```

Also, if anyone here try to do it on its own instance, once you change this in **GitLabClient.js**, you can build your minified version of the code by going to **etc/build** and launching **ant** (you'll need a java runtime and ant installed) 

Finally, there is also a bug on gitlab which makes the oAuth process fail with "The authorization server encountered an unexpected condition which prevented it from fulfilling the request.". In that case, edit you application in gitlab to remove the "confidential" flag. see https://gitlab.com/gitlab-org/gitlab/-/issues/299240 

#WorksForMeNow <3

**Comment by davidjgraph** [comment]
Fixed in 14.6.2
